moto/awslambda/models.py

```diff
@@ -489,9 +489,16 @@ def __init__(
                 ).hexdigest()
                 self.code_size = 0
             else:
-                uri, tag = self.code["ImageUri"].split(":")
+                if "@" in self.code["ImageUri"]:
+                    # deploying via digest
+                    uri, digest = self.code["ImageUri"].split("@")
+                    image_id = {"imageDigest": digest}
+                else:
+                    # deploying via tag
+                    uri, tag = self.code["ImageUri"].split(":")
+                    image_id = {"imageTag": tag}
+
```

Comment:
When deploying multi-arch images it's usually pretty sane to use the `{imageUri}@{digest}` format. Also, it's just a supported method of doing things, and therefor it should be supported. Pretty simple and reliable fix methinks.